sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/QueryPlanConstraints.scala

```diff
@@ -62,11 +62,16 @@ trait ConstraintHelper {
    */
   def inferAdditionalConstraints(constraints: Set[Expression]): Set[Expression] = {
     var inferredConstraints = Set.empty[Expression]
+    val binaryComparisons = constraints.filter(_.isInstanceOf[BinaryComparison])
```

Comment:
do you know what constraints are not `BinaryComparison`? I think it's possible, but I can't find some examples.

Reply:
Sorry. Only `BinaryComparison` is incorrect. for example: `int_column = long_column where long_column in (1L, 2L)`.

Reply:
The issue fix by https://github.com/apache/spark/pull/27252/commits/420058bf35371ddb49c67ff3c893add3281aeba9.